crates/kornia-io/src/png.rs

```diff
@@ -140,6 +199,40 @@ fn read_png_impl(file_path: impl AsRef<Path>) -> Result<(Vec<u8>, [usize; 2]), I
     Ok((buf, [info.width as usize, info.height as usize]))
 }
 
+// Utility function to decode png files from raw bytes
+fn decode_png_impl<const C: usize>(
+    src: &[u8],
+    dst: &mut [u8],
+    image_size: ImageSize,
+) -> Result<(), IoError> {
+    let mut reader = Decoder::new(src)
+        .read_info()
+        .map_err(|e| IoError::PngDecodeError(e.to_string()))?;
+
+    let image_info = reader.info();
+    if image_info.size() != (image_size.width as u32, image_size.height as u32) {
+        return Err(IoError::PngDecodeError(format!(
+            "The Image shape didn't matched. Expected H: {}, W: {}, but found H: {}, W: {}",
+            image_info.height, image_info.width, image_size.height, image_size.width
+        )));
+    }
+
+    if dst.len() < reader.output_buffer_size() {
+        return Err(IoError::PngDecodeError(format!(
```

Comment:
this is not a decoding error at all. Same as before, maybe have a dedicated error